server/command_hooks.go

```diff
@@ -522,6 +534,50 @@ func (p *Plugin) crash() {
 	_ = 1 / y
 }
 
+func (p *Plugin) executeCommandMentions(args *model.CommandArgs) *model.CommandResponse {
+	message := "The command `" + args.Command + "` contains the following different mentions.\n"
+	message += "### Mentions to users in the team\n"
+	if args.UserMentions == nil {
+		message += "_There are no mentions to users in the team in your command_.\n"
+	} else {
+		message += "| User name | Unique identifier |\n"
+		message += "|-----------|-------------------|\n"
+		for name, id := range args.UserMentions {
+			message += fmt.Sprintf("|@%s|%s|\n", name, id)
+		}
+	}
+
+	message += "\n### Mentions to public channels\n"
+	if args.ChannelMentions == nil {
+		message += "_There are no mentions to public channels in your command_.\n"
```

Comment:
From the field name it isn't clear to me that `args.ChannelMentions` only contains mentions to public channels. Could you please elaborate a bit why only mentions to public channels are resolved and why the name is `ChannelMentions`?

Reply:
The rationale behind resolving only public channels comes from the feature proposal (https://mattermost.atlassian.net/browse/MM-21987). @lieut-data, could you elaborate a bit on why we decided to resolve only public channels? I just assumed it at the moment and did not ask this to you.
In any case, @hanzei is right that it would have been more sensible to call this `PublicChannelMentions`, instead. We should open a ticket to track this, it may be a bit confusing for the end user.

Reply:
My thinking here was to match the webapp functionality -- private channels are never autocompleted, and only show as "mentions" if you happen to be in the channel. I made the assumption that we should start with privacy here, and expand later if we decide there's no privacy concerns.

Reply:
Thank you, makes total sense. In any case, I think we can make that change to the name of the field before it's released.

Reply:
Ticket opened to document the whole structure: https://mattermost.atlassian.net/browse/MM-23268. I'd mark this comment as resolved, as long as @hanzei agrees.

Reply:
The naming makes sense to me :+1: 

FYI: https://docs.mattermost.com/developer/slash-commands.html#custom-slash-command will also need an update.